go-controller/pkg/factory/factory.go

```diff
@@ -182,14 +182,13 @@ func getQueueNum(oType reflect.Type, obj interface{}) uint32 {
 	return h.Sum32() % uint32(numEventQueues)
 }
 
-// enqueueEvent adds an event to the queue. Caller must hold at least a read lock
-// on the informer.
+// enqueueEvent adds an event to the queue.
 func (i *informer) enqueueEvent(oldObj, obj interface{}, processFunc func(*event)) {
 	i.RLock()
-	defer i.RUnlock()
-	queueIdx := getQueueNum(i.oType, obj)
-	if i.events[queueIdx] != nil {
-		i.events[queueIdx] <- &event{
+	queue := i.events[getQueueNum(i.oType, obj)]
+	i.RUnlock()
+	if queue != nil {
+		queue <- &event{
```

Comment:
This races with shutdown and could result in you sending to a closed queue.

Reply:
Yeah, you're right. This needs a rethink.